File: javascript/rna-transcription/rna-transcription.js
function DnaTranscriber() {
    var map = {
        C: 'G',
        G: 'C',
        A: 'U',
        T: 'A'
    };

    this.toRna = function(dna) {
        return dna.split('').map(x => map[x]).join('');
    };
}

module.exports = DnaTranscriber;

File: cpp/rna-transcription/rna_transcription.cpp
#include <string>
#include <map>
#include "rna_transcription.h"


namespace transcription
{

static const std::map<char, char> _N_MAP
{
    {'G', 'C'},
    {'C', 'G'},
    {'A', 'U'},
    {'T', 'A'}
};

char to_rna(const char& dna)
{
    return _N_MAP.at(dna);
}

std::string to_rna(std::string dna)
{
    for (auto& c : dna)
    {
        c = to_rna(c);
    }
    return dna;
}


} // transcription